substrate/frame/support/src/inherent.rs

```diff
@@ -81,6 +82,13 @@ pub trait ProvideInherent {
 		Ok(())
 	}
 
+	/// The order of the inherent relative to all other inherents.
+	///
+	/// A return value of `None` indicates that the pallet index should be used.
+	fn inherent_order() -> Option<InherentOrder> {
+		None
```

Comment:
I think that defining the order at pallet level is not good design. It's up to the runtime to define the order of inherents, because for the same pallet set, different runtimes will have different requirements.